pinot-common/src/main/java/org/apache/pinot/common/utils/ZkStarter.java

```diff
@@ -52,6 +52,10 @@ private ZookeeperInstance(PublicZooKeeperServerMain serverMain, String dataDirPa
     @Override
     public void initializeAndRun(String[] args)
         throws QuorumPeerConfig.ConfigException, IOException {
+      // org.apache.log4j.jmx.* is not compatible under log4j-1.2-api, which provides the backward compatibility for
+      // log4j 1.* api for log4j2. In order to avoid 'class not found error', the following line disables log4j jmx
+      // bean registration for local zookeeper instance
+      System.setProperty("zookeeper.jmx.log4j.disable", "true");
```

Comment:
Do later versions of ZK work better with log4j2? Wondering what our plan should be to stop relying on log4j-1.2-api.

Reply:
I think that we should still keep `log4j-1.2-api` since some of our dependencies are directly using log4j 1.x api. Zookeeper is still on log4j 1.x so using later version won't help here. There was some effort to do migration but the pr didn't get merged and closed. https://github.com/apache/zookeeper/pull/148